src/cmp_aus.js

```diff
@@ -95,31 +95,132 @@ const reloadPage = async (page) => {
+	// --------------- BUNDLE END ---------------------------
 
+	// --------------- PAGESKIN START ---------------------------
+	log(`[TEST 4: PAGESKIN] Step start`);
 	const hasPageskin = await page.evaluate(() =>
 		document.body.classList.contains('has-page-skin'),
 	);
 
 	if (hasPageskin) {
-		log('Pageskin detected. Prebid will not run');
+		log('[TEST 4: PAGESKIN] Pageskin detected. Prebid will not run');
 		return Promise.resolve();
 	}
+	log(`[TEST 4: PAGESKIN] Step complete`);
+	// --------------- PAGESKIN END ---------------------------
 
+	// --------------- PUBMATIC START ---------------------------
+	log(`[TEST 4: PUBMATIC] Step start`);
 	const prebidURL =
 		'https://hbopenbid.pubmatic.com/translator?source=prebid-client';
 
 	await page.waitForRequest((req) => req.url().includes(prebidURL));
-	log(`Prebid check: Complete`);
+	log(`[TEST 4: PUBMATIC] Step complete`);
+	// --------------- PUBMATIC END ---------------------------
+
+	// --------------- PBJS START ---------------------------
+	log(`[TEST 4: PBJS] Step start`);
+	const hasPrebid = await page.waitForFunction(() => window.pbjs !== undefined);
+	if (!hasPrebid) {
+		logError('[TEST 4: PBJS] Prebid.js is not loaded');
+		throw new Error('Prebid.js is missing');
+	}
+	log(`[TEST 4: PBJS] Step complete`);
+	// --------------- PBJS END ---------------------------
+
+	// --------------- BID RESPONSE START ---------------------------
+	log(`[TEST 4: BID RESPONSE] Step start`);
+
+	await page.waitForFunction(
+		() => {
+			const events = window.pbjs?.getEvents() ?? [];
+			return events.find(
+				(event) =>
+					event.eventType === 'auctionInit' &&
+					event.args.adUnitCodes.includes('dfp-ad--top-above-nav'),
+			);
+		},
+		{ timeout: 10000 },
+	);
+
+	const topAboveNavBidderRequests = await page.evaluate(() => {
+		const auctionInitEvent = window.pbjs
+			?.getEvents()
+			.find(
+				(event) =>
+					event.eventType === 'auctionInit' &&
+					event.args.adUnitCodes.includes('dfp-ad--top-above-nav'),
+			);
+
+		return auctionInitEvent?.args.bidderRequests || [];
+	});
+
+	const expectedBidders = [
+		'ix',
+		'rubicon',
+		'criteo',
+		'pubmatic',
+		'ttd',
+		'adyoulike',
+		'triplelift',
+		'and',
+		'oxd',
+	];
+
+	if (topAboveNavBidderRequests.length === 0) {
+		log(
+			'[TEST 4: BID RESPONSE] Bid Response for top-above-nav is null or pbjs is not defined',
+		);
+	}
+	if (topAboveNavBidderRequests.length !== expectedBidders.length) {
+		log(
+			`[TEST 4: BID RESPONSE] Expected ${expectedBidders.length} bidders, got ${topAboveNavBidderRequests.length}`,
+		);
+	}
+
+	const theActualBidders = topAboveNavBidderRequests.map(
+		(bidder) => bidder.bidderCode,
+	);
+	log(
+		`[TEST 4: BID RESPONSE] Actual Bidders ${JSON.stringify(theActualBidders)}`,
+	);
+
+	let allMatched = true;
+
+	expectedBidders.forEach((bidder) => {
+		if (!theActualBidders.includes(bidder)) {
+			allMatched = false;
+			log(`[TEST 4: BID RESPONSE] Missing bidder: ${bidder}`);
```

Comment:
I think some of these should be logError?

Reply:
And also need to throw an error, otherwise the canary won't fail and we won't get alerted